changelog/11390.bugfix.md

```diff
@@ -0,0 +1,2 @@
+Fixes broken `2.x` regression where the expected priority order in which slot mappings are applicable is from the beginning to the end of the slot mappings list specified in the domain.
+Clarifies this implicit priority order in the docs.
```

Comment:
```suggestion
Fixes regression in which slot mappings were prioritized according to reverse order as they were listed in the domain, instead of in order from first to last, as was implicitly expected in 2.x. 
Clarifies this implicit priority order in the docs.
```